build.gradle.kts

```diff
@@ -50,3 +78,79 @@ subprojects {
         }
     }
 }
+
+/**
+ * To run detekt simply:
+ * 1. ./gradlew module:detekt for each module
+ * 2. ./gradlew detekt for whole project
+ */
+fun Project.coreDetektSetup() {
+    // Apply Plugin to sub-project
+    apply(plugin = "io.gitlab.arturbosch.detekt")
+
+    // Configure Detekt
+    detekt {
+        config = files("$rootDir/config/detekt/detekt.yml")
+        buildUponDefaultConfig = true
+        ignoredBuildTypes = listOf("release")
+        source = files(
+            io.gitlab.arturbosch.detekt.extensions.DetektExtension.DEFAULT_SRC_DIR_JAVA,
+            io.gitlab.arturbosch.detekt.extensions.DetektExtension.DEFAULT_TEST_SRC_DIR_JAVA,
+            io.gitlab.arturbosch.detekt.extensions.DetektExtension.DEFAULT_SRC_DIR_KOTLIN,
+            io.gitlab.arturbosch.detekt.extensions.DetektExtension.DEFAULT_TEST_SRC_DIR_KOTLIN,
+            // Kotlin Multiplatform
+            "src/commonMain/kotlin",
+            "src/androidMain/kotlin",
+            "src/iosMain/kotlin",
+            "src/jvmMain/kotlin",
+            "src/desktopMain/kotlin",
+            "src/jsMain/kotlin",
```

Comment:
should we also include test?

Reply:
Yes imo. They contain code too

Reply:
Okay, please add them, they follow the same nomenclature. (commonTest, etc)

Reply:
Done